### PR TITLE
Feature: Hotbar

### PR DESCRIPTION
**Pull request by AwesomestCode**
# Getting Started on Inventories

This will be the first pull request in a series of pull requests adding support for various inventories.

## Roadmap:
1. Make the hotbar *(this pull request)*
    - [x] Get the hotbar to render properly
    - [x] Get the hotbar to show items on login
    - [x] Get the hotbar to update live
    - [x] Allow switching of the selected slot with indicators on which slot you are on.
    - [x] Allow dropping of items
2. Survival inventory
3. Chests
4. Furnaces
5. Crafting Tables
6. Villagers
7. All other containers.
8. Creative inventory

## Progress

For this pull request: 5/5 steps have finished. Currently cleaning up commits and code.
This section will be edited with every significant advancement until this pull request is closed, when it will be replaced with a link to the next PR.

## Notes
* *This series of PRs may take a while to finish, as they implement a very large feature.*
* *The earlier PR (#41) accidentally got closed; this PR is a recreation of that one.*
* *I need a map of item to texture names, some items have different names from their textures, in which case this PR doesn't work properly... yet.*

### CODE REVIEW

**Comment by AwesomestCode** [comment]
Finished step 3.

File: index.js [hunk 1]
```diff
@@ -197,6 +220,27 @@ async function main () {
...
+
+      // eslint-disable-next-line no-undef
+      const http = new XMLHttpRequest()
+      let url = newItem ? window.location.href + 'textures/1.16.4/items/' + newItem.name + '.png' : ''
+      http.open('HEAD', url)
+
+      http.onreadystatechange = function () {
+        if (this.readyState === this.DONE) {
+          if (this.status === 404) {
+            url = newItem ? window.location.href + 'textures/1.16.4/blocks/' + newItem.name + '.png' : ''
```
Comment (rom1504) [inline index.js:237]: please avoid hardcoding the version

Comment (AwesomestCode) [reply index.js]: fixed

**Comment by AwesomestCode** [comment]
I've forced pushed the commits to have a cleaner history (that ways squashing isn't necessary). This should be done. Marking as ready for review.

**Comment by AwesomestCode** [comment]
Note: dropping items means dropping the entire stack. This seems to be caused by a limitation in mineflayer that I can't fix.

**Review by Karang — APPROVED** [review]

**Comment by rom1504** [comment]
As mentioned in chat, you can use clickWindow to implementing tossing of a single item from a single slot correctly

**Comment by Karang** [comment]
> As mentioned in chat, you can use clickWindow to implementing tossing of a single item from a single slot correctly

This can be done in an other PR.

File: index.html [hunk 2]
```diff
@@ -4,7 +4,24 @@
     <title>Prismarine Web Client</title>
     <link rel="stylesheet" href="styles.css">
   </head>
-  <body id="main-body">
+  <body>
+    <div id="hotbar-wrapper">
+      <img id="hotbar-image" src="textures/1.16.4/gui/widgets.png"> <!--NOTE: DELETE THE PUBLIC HERE BEFORE MERGING! THIS IS FOR TESTING W/O WEBPACK !-->
+      <img id="hotbar-highlight" src="textures/1.16.4/gui/widgets.png">
+      <div id="hotbar-items-wrapper">
+        <img class="hotbar-item" id="hotbar-0" src="textures/1.16.4/blocks/stone.png">
```
Comment (rom1504) [inline index.html:12]: why are items hardcoded here ?

Comment (AwesomestCode) [reply index.html:12]: It's to test without Webpack. It's replaced first thing upon spawn, before the loading screen is removed, so it shouldn't be an issue.

**Review by rom1504 — APPROVED** [review]

**Pull request merged**